### PR TITLE
Fix performance issue with product_type filter

### PR DESCRIPTION
**Pull request by CossackDex**
I want to merge this change because it fix performance issue with product_type filter 

Search filter 
Before -> https://explain.dalibo.com/plan/Oce
After -> https://explain.dalibo.com/plan/Cmn

For Configurable and ProductType there's already index search.
Configurable -> https://explain.dalibo.com/plan/MU6
ProductType -> https://explain.dalibo.com/plan/9fo

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-3312 Refactor products filtering(except ProductFilter).](https://app.clickup.com/t/2549495/SALEOR-3312)

File: saleor/graphql/product/filters.py [hunk 1]
```diff
@@ -667,6 +665,13 @@ class Meta:
         model = ProductType
         fields = ["search", "configurable", "product_type"]
 
+    @classmethod
+    def filter_product_type_searchable(cls, queryset, _name, value):
```
Comment (L3str4nge) [inline saleor/graphql/product/filters.py:669]: Would be good to cover this method

Comment (CossackDex) [reply saleor/graphql/product/filters.py:669]: This case is already tested by test_product::test_filter_product_types_by_custom_search_value, if I understand you comment properly.

File: saleor/graphql/product/filters.py [hunk 2]
```diff
@@ -667,6 +665,13 @@ class Meta:
...
+        if not value:
+            return queryset
+        name_slug_qs = Q(name__ilike=value) | Q(slug__ilike=value)
+        return queryset.filter(name_slug_qs)
```
Comment (IKarbowiak) [inline saleor/graphql/product/filters.py:673]: We usually add the filter method outside filter calss.

File: saleor/graphql/product/filters.py (hunk 1 above)
Comment (L3str4nge) [reply saleor/graphql/product/filters.py:669]: Sure, I agreed. I meant a test case to see if the correct queryset is returned. Pure unit test for new added method `filter_product_type_searchable`

**Review by IKarbowiak — APPROVED** [review]

File: saleor/product/migrations/0148_auto_20210929_1143.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    operations = [
+        migrations.RemoveIndex(
+            model_name="producttype",
+            name="producttype_p_meta_idx",
+        ),
+        migrations.RemoveIndex(
+            model_name="producttype",
+            name="producttype_meta_idx",
+        ),
```
Comment (fowczarek) [inline saleor/product/migrations/0148_auto_20210929_1143.py:21]: We shouldn't remove those indexes. :police_car:

**Review by fowczarek — CHANGES_REQUESTED** [review]

**Review by L3str4nge — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

**Pull request merged**